Adding mcpl

tkittel opened the pull request:
Packaging the MCPL project which provides tools for the .mcpl format, the most widely used interchange format for particle data between Monte Carlo-based particle simulation applications (supporting 11 applications currently with a few more in the pipeline). More info: [website](https://mctools.github.io/mcpl/), [github](https://github.com/mctools/mcpl/), and [publication](https://doi.org/10.1016/j.cpc.2017.04.012).

<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Gitter channel][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

tkittel commented:
@conda-forge/staged-recipes ready for review here

ocefpaf commented on recipes/mcpl/meta.yaml:
```diff
@@ -0,0 +1,68 @@
+  version: 1.5.0
+
+source:
+  url: https://github.com/mctools/mcpl/archive/v1.5.0.tar.gz
```
```suggestion
  url: https://github.com/mctools/mcpl/archive/v{{ version }}.tar.gz
```

ocefpaf commented on recipes/mcpl/meta.yaml:
```diff
@@ -0,0 +1,68 @@
+package:
+  name: mcpl
+  version: 1.5.0
```
```suggestion
  version: {{ version }}
```

ocefpaf commented on recipes/mcpl/meta.yaml:
```diff
@@ -0,0 +1,68 @@
+package:
```
{% set version = "1.5.0" %}

ocefpaf commented on recipes/mcpl/meta.yaml:
```diff
@@ -0,0 +1,68 @@
+
+test:
+  requires:
+    - python
+    - numpy
+    - cmake
+    - zlib
```
You don't need to specify run dependencies for test b/c they must be there.
BTW, is cmake really needed here?

```suggestion
    - cmake
```

tkittel replied:
Yes, cmake is needed since one of the tests is verifying that `find_package(MCPL)` works out of the box after the mcpl conda package has been installed.

ocefpaf commented on recipes/mcpl/build.sh:
```diff
@@ -0,0 +1,52 @@
+    ${CMAKE_ARGS}
+
+make -j${CPU_COUNT:-1}
+make install
```
Is there a `make check` we can run here?

tkittel commented:
Thanks a lot for the review @ocefpaf! I essentially committed all of your suggested changes directly.

Concerning the question about "make check", there is not currently any such functionality offered by MCPL, but I have added a comment in the build.sh file, reminding me to add it if it appears.

Concerning the dependency on `cmake` during the `tests` stage, it is because I am testing that `find_package(MCPL)` works out of the box in a conda environment (it did not work initially, so the test is useful to me to have in to guard against regressions).

So I believe I have addressed all of your requested changes :-)

tkittel reviewed (commented):
Should be ok now
